File: 전화번호목록.py
# def solution(phone_book): # 합계: 83.3 / 100.0
#     answer = True
#     phone_book.sort() # 119, 97674223, 1195524421
    
#     for i in range(len(phone_book)):
#         if i != 0:
#             for j in range(len(phone_book[i])): # 8
#                 phone = phone_book[i] # 97674223
#                 if phone_book[0] == phone[:j]:
#                     answer = False 
#                     break

#     return answer


def solution(phone_book):
    answer = True
    phone_book.sort()
    
    for i in range(len(phone_book)-1): # 0, 1
        if phone_book[i+1] is not None:
            if phone_book[i] == phone_book[i+1][:len(phone_book[i])]: # 접두어이기 때문에
                answer = False
                break
            
    return answer

File: Programmers/dnjfqhd12345/week_0/전화번호목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    // 최소의 길이를 먼저 구하기
    vector<string>::iterator it;
    sort(phone_book.begin(), phone_book.end(), less<string>()); // 정렬

    for (it = phone_book.begin(); it < phone_book.end()-1; it++) {
        if ((*(it+1)).find(*(it)) != string::npos) {
            answer = false;
            break;
        }
    }
    // 이거 작은 size vector에 저장해서 일일이 비교하면 될 것 같음!
    // 정렬하고 앞 뒤 비교하면 될 것 같음.

    return answer;
}